### PR TITLE
PLT-540:Flattened VMM TF Schema

### PR DESCRIPTION
**Pull request by SivaanandM**

### CODE REVIEW

File: examples/resources/spectrocloud_virtual_machine/provider.tf [hunk 1]
```diff
@@ -12,4 +12,5 @@ provider "spectrocloud" {
   api_key      = var.sc_api_key
   project_name = var.sc_project_name
   trace        = true
+  ignore_insecure_tls_error = true
```
Comment (nikchern) [inline examples/resources/spectrocloud_virtual_machine/provider.tf:15]: please revert unnecessary changes.

File: spectrocloud/kubevirt/schema/virtualmachineinstance/template_spec.go [hunk 2]
```diff
@@ -30,34 +30,26 @@ func VirtualMachineInstanceTemplateSpecSchema() *schema.Schema {
...
-func ExpandVirtualMachineInstanceTemplateSpec(virtualMachine []interface{}) (*kubevirtapiv1.VirtualMachineInstanceTemplateSpec, error) {
-	if len(virtualMachine) == 0 || virtualMachine[0] == nil {
-		return nil, nil
-	}
-
+func ExpandVirtualMachineInstanceTemplateSpec(d *schema.ResourceData) (*kubevirtapiv1.VirtualMachineInstanceTemplateSpec, error) {
 	result := &kubevirtapiv1.VirtualMachineInstanceTemplateSpec{}
 
-	in := virtualMachine[0].(map[string]interface{})
+	// we have removed metadata for template hence trying to apply same metadata (TBD)***
```
Comment (nikchern) [inline spectrocloud/kubevirt/schema/virtualmachineinstance/template_spec.go:36]: Is this decided? Or it's still TBD?

Comment (SivaanandM) [reply spectrocloud/kubevirt/schema/virtualmachineinstance/template_spec.go:36]: Its still need to be discussed

File: examples/resources/spectrocloud_virtual_machine/provider.tf (hunk 1 above)
Comment (SivaanandM) [reply examples/resources/spectrocloud_virtual_machine/provider.tf]: removed ignore_insecure_tls_error param

File: spectrocloud/kubevirt/schema/virtualmachine/virtual_machine.go [hunk 3]
```diff
@@ -3,20 +3,85 @@ package virtualmachine
...
 			Type:        schema.TypeString,
 			Required:    true,
 			ForceNew:    true,
 			Description: "The cluster UID to which the virtual machine belongs to.",
 		},
+		"cluster_context": {
+			Type:         schema.TypeString,
+			Optional:     true,
+			Default:      "project",
+			ValidateFunc: validation.StringInSlice([]string{"", "project", "tenant"}, false),
```
Comment (nikchern) [inline spectrocloud/kubevirt/schema/virtualmachine/virtual_machine.go:83]: Need to add description.

File: spectrocloud/kubevirt/schema/virtualmachine/spec.go [hunk 4]
```diff
@@ -84,22 +34,120 @@ func expandVirtualMachineSpec(virtualMachine []interface{}) (kubevirtapiv1.Virtu
...
 
 	return []interface{}{att}
 }
+
+func FlattenVMMToSpectroSchema(in kubevirtapiv1.VirtualMachineSpec, resourceData *schema.ResourceData) error {
+	VM := FlattenVirtualMachineSpec(in, resourceData)[0].(map[string]interface{})
+	VMTemplate := VM["template"]
+	VMTemplateSpec := VMTemplate.([]interface{})[0].(map[string]interface{})["spec"]
+	VMTemplateSpecAttributes := VMTemplateSpec.([]interface{})[0].(map[string]interface{})
+	if err := resourceData.Set("run_strategy", VM["run_strategy"]); err != nil {
```
Comment (nikchern) [inline spectrocloud/kubevirt/schema/virtualmachine/spec.go:62]: need to bring it common style.

Comment (SivaanandM) [reply spectrocloud/kubevirt/schema/virtualmachine/spec.go:76]: For now added the comments and organized into mandatory and nonmandatory attributes

**Review by nikchern — APPROVED** [review]
